### PR TITLE
Added submission to ona (again).

### PR DESCRIPTION
**Pull request by bruskiza**
WIP.

### CODE REVIEW

File: go-app.js [hunk 1]
```diff
@@ -232,14 +234,96 @@ go.app = function() {
...
+            auth: {
+              username: ona_conf.username,
+              password: ona_conf.password
+            },
+            url: ona_conf.url
+          });
+          var data = self.im.user.answers;
+          data.create_date_time = new Date();
+          data.reported_by = self.im.user.addr;
+          data.gender = (self.im.user.answers.No_SA_ID_Gender_Entry == 1) ? "male" : "female";
```
Comment (smn) [inline go-app.js:257]: I think there's a subtle bug here, the gender's state value is actually stored as a string not as an integer.
You can change the ChoiceState's `Choice` value to be `male` or `female`, that way you don't have to do this switch. The ChoiceState automatically maps the `1` & `2` entries to those values.

Comment (bruskiza) [inline go-app.js:257]: I know.. pull request https://github.com/praekelt/go-jsbox-malaria24/pull/21 which i am going to write a test for and practice merging again...

File: go-app.js [hunk 2]
```diff
@@ -232,14 +234,96 @@ go.app = function() {
...
+          said = data.SA_ID_Entry;
+          if (data.SA_ID_Entry) {
+            data.No_SA_ID_Year_Entry = said.substring(0, 2);
+            data.No_SA_ID_Month_Entry = said.substring(2, 4);
+            data.No_SA_ID_Day_Entry = said.substring(4, 6);
+          }
```
Comment (smn) [inline go-app.js:263]: Can you explain what's going on here? What's the relation between `SA_ID_Entry` and year, month & day? Is that information encoded in the SA ID?

File: go-app.js (hunk 1 above)
Comment (bruskiza) [inline go-app.js:257]: So please +1 this and we will fix in PR21.

File: go-app.js (hunk 2 above)
Comment (bruskiza) [inline go-app.js:263]: Correct... I could have done it with a sprintf function or something but #time

Comment (smn) [inline go-app.js:263]: No substrings is fine. I just wasn't aware that that's what's encoded in there.

**Comment by smn** [comment]
:+1:

**Pull request merged**